### ISSUE TITLE
[Custom UI5 Web Component]:

### ISSUE THREAD

**Issue by akshilverma** [SAP/ui5-webcomponents#7225]
### Bug Description

I have created a custom webcomponent using ui5-webcomponent framework. This custom webcomponent is then used in a frontend project based on React.

Our custom webcomponent currently uses v1.12.0. I am trying to upgrade the ui5-webcomponent’s version from v1.12.0 to v1.14.1 for my custom webcomponent.

When I run the custom webcomponent locally with the upgraded version (using npm start), it starts up normally. However, when I try to integrate the webcomponent with our React app, I get an error - "Uncaught (in promise) TypeError: Cannot read properties of undefined" - for all the properties/attributes defined under the properties key of the metadata object.

I have noticed that this problem started with v1.13.0. Is there something new that was introduced which I might have missed?

### Affected Component

Custom UI5 Web Component

### Expected Behaviour

The custom webcomponent's properties should not be undefined.

### Log Output, Stack Trace or Screenshots

<img width="631" alt="Screenshot 2023-06-19 at 14 18 54" src="https://github.com/SAP/ui5-webcomponents/assets/101528886/4ade5977-1373-41c3-840c-b607879184b3">

### Priority

None

### UI5 Web Components Version

1.13.x, 1.14.x

### Browser

Chrome

### Operating System

MacOS

### Organization

SAP

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by unazko** [comment]
Hello @akshilverma,

We would need a short sample showcasing the custom control usage and the issue like the following:
https://codesandbox.io/s/71r1x5o51q?fontsize=14&module=%2Findex.html

Best regareds,
Boyan

**Comment by ilhan007** [comment]
Hi colleagues, @akshilverma 
I assume it might be difficult to recreate the issue in the given [codesandbox](https://codesandbox.io/s/71r1x5o51q?fontsize=14&module=%2Findex.html) above.

Could you at least test, if you have the same issue when the custom web component (sap-build-action-create) is used in a fresh getting started React app, generated  by `npx create-next-app`.

**Comment by akshilverma** [comment]
Hi Ilhan,

I'm not very well versed in nextjs. When I try to integrate the webcomponent with a new next app, I get a different error here: _ReferenceError: HTMLElement is not _defined__.

I'm not sure if there is some extra setting required here for a nextjs app to run a custom web component.

<img width="1363" alt="Screenshot 2023-06-20 at 17 46 38" src="https://github.com/SAP/ui5-webcomponents/assets/101528886/03d5ea97-545e-45c4-97e3-1d0c8ad4c409">

**Comment by ilhan007** [comment]
Hello colleagues,  @akshilverma 

we investigated and resolved the issue offline - the root cause was mainly about having different versions of ui5/webcomponents-* installed, from one side by the React app and the custom web component itself. 
There were 1.12.* and 1.14.* versions installed causing wrong context being set in the templates, leading to the exception, shown in the description. 

Once we arranged and installed all versions properly, the custom web component rendered correctly in the React app.
